transports/ftp/src/test/java/org/mule/transport/ftp/FtpConnectionSocketTimeoutTestCase.java

```diff
@@ -161,14 +160,11 @@ public FtpletResult onConnect(FtpSession session) throws FtpException, IOExcepti
 
     private static void sleep()
     {
-        try
+        long now = currentTimeMillis();
+        long stopTime = now + SERVER_SLEEP;
+        while (now < stopTime)
```

Comment:
When I requested to avoid Thread.sleep I mean to use a latch or something that does not involves wasting time. Here you are using busy waiting, which is worst than the sleep because it has all the same problems and it also spends additional processing time